tests/simplex/step-01.cc

```diff
@@ -0,0 +1,105 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Step-01 on a simplex mesh. Following incompatible modifications had to be
+// made:
+//  - Replace or convert triangulation to a simplex based triangulation.
+//  - Copy manifolds after conversion to simplex based triangulation in order
+//    to execute refinement.
+
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+
+#include <deal.II/simplex/grid_generator.h>
+
+#include <cmath>
+#include <fstream>
+#include <iostream>
+
+#include "../tests.h"
+
+using namespace dealii;
+
+void
+first_grid()
+{
+  Triangulation<2> triangulation;
+  GridGenerator::subdivided_hyper_cube_with_simplices(triangulation, 1);
+  triangulation.refine_global(2);
+
+  GridOut grid_out;
+  grid_out.write_svg(triangulation, deallog.get_file_stream());
+  deallog << "Grid written to grid-1.svg" << std::endl;
+}
+
+
+
+void
+second_grid()
+{
+  Triangulation<2> tria_temp, triangulation;
+
+  const Point<2> center(1, 0);
+  const double   inner_radius = 0.5, outer_radius = 1.0;
+  GridGenerator::hyper_shell(tria_temp, center, inner_radius, outer_radius, 10);
+
+  // convert mesh to a simplex mesh and copy manually the attached manifolds
+  GridGenerator::convert_hypercube_to_simplex_mesh(tria_temp, triangulation);
+  for (const auto i : tria_temp.get_manifold_ids())
+    if (i != numbers::flat_manifold_id)
+      triangulation.set_manifold(i, tria_temp.get_manifold(i));
```

Comment:
I did see #11389 but this code in particular makes it look like this should work with manifolds right now.

Reply:
Oh, I see - we copy the manifold but not the manifold id. Got it!

Reply:
Yeah. This is an issue that I really don't like, since people (in particular me) tend to forget to copy the actual manifolds.